### ISSUE TITLE
`TypeError: ShortCommitHash requires length of 7`

### ISSUE THREAD

**Issue by PawelLipski** [PawelLipski/git-machete#356]
As reported by @kgadek:
```
Traceback (most recent call last):
  File "/usr/local/bin/git-machete", line 19, in <module>
    cli.main()
  File "/usr/local/Cellar/git-machete/3.5.0/libexec/lib/python3.9/site-packages/git_machete/cli.py", line 813, in main
    launch(sys.argv[1:])
  File "/usr/local/Cellar/git-machete/3.5.0/libexec/lib/python3.9/site-packages/git_machete/cli.py", line 588, in launch
    machete_client.set_fork_point_override(branch, upstream)
  File "/usr/local/Cellar/git-machete/3.5.0/libexec/lib/python3.9/site-packages/git_machete/client.py", line 1618, in set_fork_point_override
    f"{self.__git.get_revision_repr(to_revision)}</b>.\n",
  File "/usr/local/Cellar/git-machete/3.5.0/libexec/lib/python3.9/site-packages/git_machete/git_operations.py", line 567, in get_revision_repr
    short_sha = self.get_short_commit_sha_by_revision(revision)
  File "/usr/local/Cellar/git-machete/3.5.0/libexec/lib/python3.9/site-packages/git_machete/git_operations.py", line 346, in get_short_commit_sha_by_revision
    self.__short_commit_sha_by_revision_cached[revision] = self.__find_short_commit_sha_by_revision(revision)
  File "/usr/local/Cellar/git-machete/3.5.0/libexec/lib/python3.9/site-packages/git_machete/git_operations.py", line 342, in __find_short_commit_sha_by_revision
    return ShortCommitHash.of(self._popen_git("rev-parse", "--short", revision + "^{commit}").rstrip())
  File "/usr/local/Cellar/git-machete/3.5.0/libexec/lib/python3.9/site-packages/git_machete/git_operations.py", line 132, in of
    raise TypeError(f'ShortCommitHash requires length of 7. Given value: {value}, has length: {len(value)}.\n'
TypeError: ShortCommitHash requires length of 7. Given value: 03e76983, has length: 8.
Consider posting an issue on https://github.com/VirtusLab/git-machete/issues/new
```

**Comment by PawelLipski** [comment]
Ofc short commit hash can be longer, since `git rev-parse --short` returns the shortest **unambiguous** prefix (but at least 7 hex digits)

**Comment by PawelLipski** [comment]
Rel: https://virtuslab.github.io/tips/#other/birthday-paradox

**Comment by rifel123** [comment]
closed with #357
